homeassistant/components/axis/binary_sensor.py

```diff
@@ -104,6 +108,21 @@ def name(self):
                 f"{self.device.name} {self.device.api.vapix.ports[self.event.id].name}"
             )
 
+        if self.event.CLASS == CLASS_MOTION:
+
+            for event_class, event_data in (
+                (FenceGuard, self.device.api.vapix.fence_guard),
+                (LoiteringGuard, self.device.api.vapix.loitering_guard),
+                (MotionGuard, self.device.api.vapix.motion_guard),
+                (Vmd4, self.device.api.vapix.vmd4),
+            ):
+                if (
+                    isinstance(self.event, event_class)
+                    and event_data
+                    and self.event.id in event_data
+                ):
+                    return f"{self.device.name} {self.event.TYPE} {event_data[self.event.id].name}"
```

Comment:
Should we mark this a breaking change in case anyone was matching for specific text?

Reply:
True, Will do!